certification/internal/shell/scorecard_olm_suilte.go

```diff
@@ -0,0 +1,122 @@
+    "github.com/komish/preflight/certification"
+    "github.com/sirupsen/logrus"
+)
+
+type ScorecardOlmSuiteCheck struct{}
+
+func (p *ScorecardOlmSuiteCheck) Validate(bundle string, logger *logrus.Logger) (bool, error) {
+    
+    //Currentdir
+    homedir, err := os.Getwd()
+    if err != nil {
+        logger.Error("unable to get current directory: ", err)
+        return false, err
+    }
+    
+	artifactsdir := filepath.Join(homedir,"/artifacts")
+	err = os.MkdirAll(artifactsdir, 0777)
+    if err != nil {
+        logger.Error("unable to create artifactsdir: ", err)
+        return false, err
+    }
+    
+	scorecardresult := "operator_bundle_scorecard_olmSuiteCheck.json"
+    
+    tmpdir, err := os.MkdirTemp("", "preflight")
+    if err != nil {
+        logger.Error("unable to create TempDir: ", err)
+        return false, err
+    }
+    defer os.RemoveAll(tmpdir)
+    bundledir := filepath.Join(tmpdir,"/bundle")
+    
+    err = os.MkdirAll(bundledir, 0777)
+    if err != nil {
+        logger.Error("unable to create subdirs: ", err)
+        return false, err
+    }
+
+    stdouterr, err := exec.Command("oc", "image", "extract", "--path", "/:"+bundledir, bundle).CombinedOutput()
+    if err != nil {
+        logger.Error("unable to execute oc image extract on the image: ", err)
+        return false, err
+    }
+
+    // Need to be in the parent directory of the bundle contents
+    err = os.Chdir(tmpdir)
+    if err != nil {
+        logger.Error("unable to change dir: ", err)
+        return false, err
+    }
+    
+    configfile := filepath.Join(bundledir,"/tests/scorecard/config.yaml")
+
+    //copy config file to the root
+    input, err := ioutil.ReadFile(configfile)
+    if err != nil {
+        logger.Error("unable to read config.yaml: ", err)
+        return false, err
+    }
+    
+    destinationfile := filepath.Join(tmpdir,"/config.yaml")
+    err = ioutil.WriteFile(destinationfile, input, 0777)
+    if err != nil {
+        logger.Error("unable to write to config.yaml file: ", err)
+        return false, err
+    }
+    
+    _, err = exec.Command("chmod", "-R", "go+r", "./").CombinedOutput()
+    if err != nil {
+        logger.Error("unable to execute chmod: ", err)
+        return false, err
+    }
+    
+    logger.Debugf("Running Scorecard Check for ",bundle)
+    stdouterr, scorecarderr := exec.Command("operator-sdk", "scorecard",   
+                                  "--config", destinationfile,
+								  "--selector=suite=olm",
+                                  "--output", "json", "bundle").CombinedOutput()
+    
+    scorecardfile := filepath.Join(artifactsdir,"/",scorecardresult)
+    err = ioutil.WriteFile(scorecardfile, stdouterr, 0644)
+    if err != nil {
+        logger.Error("unable to copy result to /artifacts subdir: ", err)
+        return false, err
+    }
+
+	if scorecarderr != nil {
+        logger.Error("unable to execute operator-sdk scorecard: ", err)
+        return false, err
+    }
+    
```

Comment:
I feel you should do a check on the output file to ensure all tests pass. You can search for `"state": "pass"` in the output file to return `true,nil` or `false,nil`

Reply:
I added guardrails against them.